channels/OFF-CHAIN.md

```diff
@@ -0,0 +1,503 @@
+- `lock_period` SHOULD be sufficient time to safely publish transactions to the
+  blockchain to stop a cheater
+- `responder_pubkey` MUST be a valid ed25519 pubkey
+
+
+### `funding_created`
+
+In order to open a channel on chain, peers need to cooperate and co-sign the
+`channel_create` transaction. The `funding_created` message is used by the
+initiator to send their signature to the responder.
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| transaction_data     | 32 |
+ ---------------------- ----
+| signature            | 64 |
+ ---------------------- ----
+```
+
+
+#### Requirements
+
+*Responder*:
+
+- MUST abort if the signature is invalid for the provided transaction data
+
+
+### `funding_signed`
+
+If the responder was able to validate the initiator's signature sent in the
+`funding_created` message, then it should proceed to respond with its
+`funding_signed` message, unless the initiator specified a `push_amount > 0` in
+the `channel_open` message. In the latter case, they should wait for the
+initiator to send the initial state signed by them.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| signature            | 64 |
+ ---------------------- ----
+```
+
+
+### `funding_locked`
+
+Opening a channel requires an on-chain transaction. This transaction needs to be
+included in a block and, since we only have probabilistic finality, be
+sufficiently confirmed, s.t. the probability of a chain re-organisation is
+negligible.
+
+This message is exchanged by both peers to signal to each other that the above
+condition has been met from their point of view and only after both of them
+agree on this, can the channel be considered to be opened.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| temporary_channel_id | 32 |
+ ---------------------- ----
+| transaction_id       | 32 |
+ ---------------------- ----
+```
+
+
+#### Requirements
+
+A node MUST NOT send the `funding_locked` message unless the `channel_create`
+transaction has `minimum_depth` confirmations.
+
+## Initialise contract
+
+### `update_init_contract`
+
+
+## State update
+
+State updates require consent of both peers.
+
+Each update MUST have a strictly increasing sequence number, which SHOULD start
+at `0` on channel initialisation.
+
+Parameters:
+
+- `channel_id`:
+- `balances`:
+- `state`
+
+
+### `update_exec_func`
+
+
+
+### `update_exec_result`
+
+## Deposit
+
+Depositing funds into a channel should increase the longevity of channels, since
+it makes balancing them easier but seeing as this incurs an on-chain transaction
+we'd still like to avoid it, if possible.
+
+
+### `update_deposit`
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| amount               | 8  |
```

Comment:
Shall this not have a `from` field too? Or is it implicit from signature of this state update message?